### PR TITLE
[Exporter.Geneva][Otlp protobuf] Add PrepopulatedMetricDimensions and Account/Namespace support

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Towards #1585 

## Changes

Adds support for PrepopulatedMetricDimensions options and specifying account and namespace via connection string. 

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Exporter.Geneva/Metrics/OtlpProtobuf/OtlpProtobufSerializer.cs [hunk 1]
```diff
@@ -37,9 +49,40 @@ internal sealed class OtlpProtobufSerializer
...
+            cursor = 0;
+            SerializeTags(temp, ref cursor, prepopulatedMetricDimensions, FieldNumberConstants.HistogramDataPoint_attributes);
+            this.prepopulatedHistogramDataPointAttributes = new byte[cursor];
+            Array.Copy(temp, this.prepopulatedHistogramDataPointAttributes, cursor);
+            this.prepopulatedHistogramDataPointAttributesLength = cursor;
+
+            // TODO: exponential histogram.
+        }
+
+        if (connectionStringBuilder != null && connectionStringBuilder.Namespace != null)
```
Comment (utpilla) [inline src/OpenTelemetry.Exporter.Geneva/Metrics/OtlpProtobuf/OtlpProtobufSerializer.cs:77]: Add a try/catch block as accessing `Namespace` and `Account` from `ConnectionStringBuilder` could throw an exception when connectionString does not contain them.

File: test/OpenTelemetry.Exporter.Geneva.Tests/OtlpProtobufMetricExporterTests.cs [hunk 2]
```diff
@@ -115,7 +150,7 @@ public void CounterSerializationSingleMetricPoint(string instrumentName, long? l
 
         Assert.NotNull(request.ResourceMetrics[0].Resource);
 
-        AssertOtlpAttributes([new KeyValuePair<string, object>("TestResourceKey", "TestResourceValue")], request.ResourceMetrics[0].Resource.Attributes);
+        AssertOtlpAttributes(addAccountAndNamespace ? resourceAttributes.Concat(accountAndNamespace) : resourceAttributes, request.ResourceMetrics[0].Resource.Attributes);
```
Comment (utpilla) [inline test/OpenTelemetry.Exporter.Geneva.Tests/OtlpProtobufMetricExporterTests.cs:153]: How is it testing if the account and namespace provided in the ConnectionString are also added to the resource?

Also, `AssertOtlpAttributes` should have failed. It looks like `AssertOtlpAttributes` should be updated to verify that the count of expected values matches the count of actual values.

Comment (vishweshbankwar) [reply test/OpenTelemetry.Exporter.Geneva.Tests/OtlpProtobufMetricExporterTests.cs:155]: Take a look at this e48ff46359568badac25662a1ccc932d393c6c36. I refactored it a bit to make it more clear.

> AssertOtlpAttributes should have failed. It looks like AssertOtlpAttributes should be updated to verify that the count of expected values matches the count of actual values

It does here - https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/113e5b82b06b4cb9da232f3f27547591d98d38f6/test/OpenTelemetry.Exporter.Geneva.Tests/OtlpProtobufMetricExporterTests.cs#L955

File: src/OpenTelemetry.Exporter.Geneva/Metrics/OtlpProtobuf/OtlpProtobufSerializer.cs (hunk 1 above)
Comment (vishweshbankwar) [reply src/OpenTelemetry.Exporter.Geneva/Metrics/OtlpProtobuf/OtlpProtobufSerializer.cs]: added

File: test/OpenTelemetry.Exporter.Geneva.Tests/OtlpProtobufMetricExporterTests.cs (hunk 2 above)
Comment (utpilla) [reply test/OpenTelemetry.Exporter.Geneva.Tests/OtlpProtobufMetricExporterTests.cs:155]: The tests should also check for the case where a different set of `Account` and `Namespace` are provided in the Resource by the user. Then check whether the serialized resource attributes contain both the sets of `Account` and `Namespace`.

Comment (utpilla) [reply test/OpenTelemetry.Exporter.Geneva.Tests/OtlpProtobufMetricExporterTests.cs:155]: Not related to this PR but consider updating the `AssertOtlpAttributes` method:
- There is a redundant check for attribute key.
- There is no need to count `expectedSize` in the loop. `expectedAttributesCount` should have the expected value.

**Review by utpilla — APPROVED** [review]

File: test/OpenTelemetry.Exporter.Geneva.Tests/OtlpProtobufMetricExporterTests.cs (hunk 2 above)
Comment (vishweshbankwar) [reply test/OpenTelemetry.Exporter.Geneva.Tests/OtlpProtobufMetricExporterTests.cs:157]: Done - 001a1fb5a4bd8792906abf0820630d3f141f646b

**Review by CodeBlanch — APPROVED** [review]
LGTM

**Pull request merged**